opentelemetry-api/src/opentelemetry/trace/__init__.py

```diff
@@ -658,53 +653,35 @@ def get_tracer(
     This function is a convenience wrapper for
     opentelemetry.trace.tracer_provider().get_tracer
     """
-    return tracer_provider().get_tracer(
+    return get_tracer_provider().get_tracer(
         instrumenting_module_name, instrumenting_library_version
     )
 
 
-def tracer_provider() -> TracerProvider:
-    """Gets the current global :class:`~.TracerProvider` object.
-
-    If there isn't one set yet, a default will be loaded.
-    """
-    global _TRACER_PROVIDER, _TRACER_PROVIDER_FACTORY  # pylint:disable=global-statement
+def get_tracer_provider() -> TracerProvider:
+    """Gets the current global :class:`~.TracerProvider` object."""
+    global _TRACER_PROVIDER  # pylint: disable=global-statement
 
     if _TRACER_PROVIDER is None:
-        # pylint:disable=protected-access
+        configured_tracer_provider = (
+            Configuration().tracer_provider  # pylint: disable=no-member
+        )
+
         try:
-            _TRACER_PROVIDER = loader._load_impl(
-                TracerProvider, _TRACER_PROVIDER_FACTORY  # type: ignore
+            _TRACER_PROVIDER = next(
+                iter_entry_points(
+                    "opentelemetry_tracer_provider",
+                    name=configured_tracer_provider,
+                )
+            ).load()()
+        except Exception:  # pylint: disable=broad-except
+            # FIXME Decide on how to handle this. Should an exception be
+            # raised here, or only a message should be logged and should
+            # we fall back to the default tracer provider?
+            _LOGGER.error(
+                "Failed to load tracer implementation: %s",
+                configured_tracer_provider,
```

Comment:
I would keep the logic to warn and fall back to the `DefaultTracerProvider()` here

Reply:
If we fall back to `DefaultTracerProvider` the user would likely be getting a later `AttributeError` when the tracer provider does not have an `add_span_processor` attribute. I think it is better to fail as soon as possible to give the user a clearer, easier to debug error instead of carrying on and providing an error with a longer traceback later.